src/expression/comparison_expression.cpp

```diff
@@ -80,5 +80,56 @@ const std::string ComparisonExpression::GetInfo() const {
   return os.str();
 }
 
+bool ComparisonExpression::SymmetricEquals(
+    const AbstractExpression &other) const {
+  PELOTON_ASSERT(children_.size() == 2);
+
+  if (other.GetChildrenSize() != 2) {
+    return false;
+  }
+
+  auto vcl = children_[0].get();
+  auto vcr = children_[1].get();
+  auto ocl = other.GetModifiableChild(0);
+  auto ocr = other.GetModifiableChild(1);
+  auto oexp_type_ = other.GetExpressionType();
+  try {
+    switch (exp_type_) {
+      case (ExpressionType::COMPARE_EQUAL):
+      case (ExpressionType::COMPARE_NOTEQUAL):
+        return exp_type_ == oexp_type_ &&
+               ((ocl->ExactlyEquals(*vcl) && ocr->ExactlyEquals(*vcr)) ||
+                (ocl->ExactlyEquals(*vcr) && ocr->ExactlyEquals(*vcl)));
+      case (ExpressionType::COMPARE_LESSTHAN):
+        return (ocl->ExactlyEquals(*vcl) && ocr->ExactlyEquals(*vcr) &&
+                oexp_type_ == ExpressionType::COMPARE_LESSTHAN) ||
+               (ocl->ExactlyEquals(*vcr) && ocr->ExactlyEquals(*vcl) &&
+                oexp_type_ == ExpressionType::COMPARE_GREATERTHAN);
+      case (ExpressionType::COMPARE_GREATERTHAN):
+        return (ocl->ExactlyEquals(*vcl) && ocr->ExactlyEquals(*vcr) &&
+                oexp_type_ == ExpressionType::COMPARE_GREATERTHAN) ||
+               (ocl->ExactlyEquals(*vcr) && ocr->ExactlyEquals(*vcl) &&
+                oexp_type_ == ExpressionType::COMPARE_LESSTHAN);
+      case (ExpressionType::COMPARE_LESSTHANOREQUALTO):
+        return (ocl->ExactlyEquals(*vcl) && ocr->ExactlyEquals(*vcr) &&
+                oexp_type_ == ExpressionType::COMPARE_LESSTHANOREQUALTO) ||
+               (ocl->ExactlyEquals(*vcr) && ocr->ExactlyEquals(*vcl) &&
+                oexp_type_ == ExpressionType::COMPARE_GREATERTHANOREQUALTO);
+      case (ExpressionType::COMPARE_GREATERTHANOREQUALTO):
+        return (ocl->ExactlyEquals(*vcl) && ocr->ExactlyEquals(*vcr) &&
+                oexp_type_ == ExpressionType::COMPARE_GREATERTHANOREQUALTO) ||
+               (ocl->ExactlyEquals(*vcr) && ocr->ExactlyEquals(*vcl) &&
+                oexp_type_ == ExpressionType::COMPARE_LESSTHANOREQUALTO);
+      case (ExpressionType::COMPARE_DISTINCT_FROM):
+        return exp_type_ == oexp_type_ && ocl->ExactlyEquals(*vcl) &&
+               ocr->ExactlyEquals(*vcr);
+      default:
+        throw Exception("Invalid comparison expression type.");
```

Comment:
if you are simply catching the exception you throw here, you should probably just return false in `default` and log information that there was an invalid comparison expression type. If you want the exception to cause a failure, then you can throw but would recommend not catching here. If you are expecting another exception from this `try` block, then I'd recommend picking a finer grained exception in the catch statement.